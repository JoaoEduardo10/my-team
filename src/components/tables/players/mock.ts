export const mockPlayers = [
  {
    player: {
      age: 18,
      id: 1,
      name: 'eduardo',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 20,
      id: 2,
      name: 'felipe',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 20,
      id: 3,
      name: 'caio',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 30,
      id: 4,
      name: 'renato',
      nationality: 'brasileiro'
    }
  },
];
