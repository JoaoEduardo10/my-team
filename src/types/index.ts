type TCountry = {
  code: string
  flag: string
  name: string
}

export type TSeason = {
  current: string
  end: string
  start: string
  year: number
}

export type TCountrys = {
  response: TCountry[]
}

export type TTeam = {
  response: [
    {
      team: {
        code: string
        country:string
        id: number
        name:string
        logo: string
      }
    }
  ]
}

export type TLeagues = {
  response: [
    {
      country: TCountry
      league: {
        id: number
        logo: string
        name: string
        type: string
      }
      seasons: TSeason[]
    }
  ]
}

export type TPlayers = {
  response: [
    {
      player: {
        age: number
        id: number
        name: string
        nationality: string
      }
    }
  ]
}

export type TSubFixtures = {
  away: number
  home: number
  total: number
}

export type TSubGoals = {
  percentage: string
  total: number
}

export type TSubLineups = {
  formation: string
  played: number
}

export type TTeamStatistic = {
  response: {
      fixtures: {
        draws: TSubFixtures
        loses: TSubFixtures
        played: TSubFixtures
        wins: TSubFixtures
      },
      goals: {
        against: {
          minute: {
            '0-15': TSubGoals
            '16-30': TSubGoals
            '31-45': TSubGoals
            '46-60': TSubGoals
            '61-75': TSubGoals
            '76-90': TSubGoals
            '91-105': TSubGoals
            '106-120': TSubGoals
        }
        }
      },
      lineups: TSubLineups[],
      team: {
        id: number
        logo: string
        name: string
      }
    }

}
