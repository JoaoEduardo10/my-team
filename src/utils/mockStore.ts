import configureStore from 'redux-mock-store';


const mockStore = configureStore([]);

const store = mockStore({
  team: {
    country: 'brasil',
    season: '2022',
    league: '47',
    team: 'SÃ£o paulo'
  }
});

export { store };
