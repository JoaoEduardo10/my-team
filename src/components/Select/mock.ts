/* eslint-disable @typescript-eslint/no-unused-vars */
import { TCountrys, TLeagues, TTeam } from '../../types';

const mockSeason: number[] = [2022, 2010, 2020];

const mockContrys:TCountrys['response'] = [
  {
    code: 'T1',
    flag: 'test',
    name:  'test 1'
  },
  {
    code: 'T2',
    flag: 'test 2',
    name:  'test 2'
  },
  {
    code: 'T3',
    flag: 'test3',
    name:  'test 3'
  },
];


const mockLeague: TLeagues['response'] = [
  {
    country: {
      code: 'T1',
      flag: 'test',
      name: 'Test 1'
    },
    league: {
      id: 1,
      logo: 'tes.png',
      name: 'test ligue',
      type: 'test'
    },
    seasons: [
      {
        current: 'test',
        end: '10/10/2022',
        start: '01/01/2022',
        year: 2022
      }
    ]
  }
];

const mockTeam: TTeam['response'] = [
  {
    team: {
      code: 'br',
      country: 'brazil',
      id: 1,
      logo: 'test.png',
      name: 'bar√ßa'
    }
  }
];

export { mockContrys, mockLeague, mockSeason, mockTeam };
