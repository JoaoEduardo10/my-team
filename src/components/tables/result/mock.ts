import { TTeamStatistic } from '../../../types';

export const mockResult: TTeamStatistic['response'] = {
  fixtures: {
    draws: {
      away: 4,
      home: 4,
      total: 8
    },
    loses:{
      away: 1,
      home: 4,
      total: 5
    },
    played: {
      away: 1,
      home: 10,
      total: 29
    },
    wins:{
      away: 1,
      home: 5,
      total: 6
    }
  },
  goals: {
    against: {
      minute: {
        '0-15': { percentage: '32.5', total: 2 },
        '106-120': { percentage: '32.5', total: 2 },
        '16-30': { percentage: '32.5', total: 2 },
        '31-45': { percentage: '32.5', total: 2 },
        '46-60': { percentage: '32.5', total: 2 },
        '61-75': { percentage: '32.5', total: 2 },
        '76-90': { percentage: '32.5', total: 2 },
        '91-105': { percentage: '32.5', total: 2 },
      }
    }
  },
  lineups: [
    {
      formation: '4-3-3',
      played: 7
    },
    {
      formation: '4-4-3',
      played: 10
    },
    {
      formation: '4-5-2',
      played: 13
    },
  ],
  team: {
    id: 12,
    logo: 'Saopaulo.png',
    name: 'SÃ£o Paulo'
  }
};
