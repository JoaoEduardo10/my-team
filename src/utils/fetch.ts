/* eslint-disable @typescript-eslint/no-explicit-any */

import { TCountrys, TLeagues, TTeam } from '../types';

const url = import.meta.env.VITE_URL_API as string;

const authUser = async (key: string) => {
  const response = await fetch(`${url}/status`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data = await response.json();

  return data;

};

const getCountry = async (key: string) => {
  const response = await fetch(`${url}/countries`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TCountrys = await response.json();

  return data;
};

const getLeague= async (key: string, country: string, season: number) => {
  const response = await fetch(`${url}/leagues?country=${country}&season=${season}`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TLeagues = await response.json();

  return data;
};

const getSeason = async (key: string) => {
  const response = await fetch(`${url}/leagues/seasons`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: { response: number[] } = await response.json();

  return data;
};

const getTeam = async (key: string, season: number, league: number) => {
  const response = await fetch(`${url}/teams?league=${league}&season=${season}`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TTeam = await response.json();

  return data;
};


export { authUser, getCountry, getLeague, getSeason, getTeam };
