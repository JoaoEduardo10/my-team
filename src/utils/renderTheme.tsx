import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import { GlobalStyles } from '../styles/globals-styles';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import { AuthProvider } from '../context/auth';
import { Provider } from 'react-redux';
import { store } from './mockStore';

const mockRouet = (children: React.ReactNode, path?: string) => {
  return createBrowserRouter([
    {
      path: path ?? '/',
      element: children
    }
  ]);
};

export const renderTheme = (
  children: React.ReactNode,
  path?: string
) => {
  return render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <AuthProvider>
          <RouterProvider router={mockRouet(children, path)} />
        </AuthProvider>
        <GlobalStyles />
      </ThemeProvider>
    </Provider>
  );
};
