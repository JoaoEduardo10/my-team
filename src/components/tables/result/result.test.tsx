import { screen } from '@testing-library/react';
import { Result } from '.';
import { renderTheme } from '../../../utils/renderTheme';
import { mockResult } from './mock';
import { TTeamStatistic } from '../../../types';

const newMock: TTeamStatistic = {
  response: {
    fixtures: {
      draws: {
        away: 0,
        home: 0,
        total: 0
      },
      loses: {
        away: 0,
        home: 0,
        total: 0
      },
      played: {
        away: 0,
        home: 0,
        total: 0
      },
      wins: {
        away: 0,
        home: 0,
        total: 0
      },
    },
    goals: {
      against: {
        minute: {
          '0-15': {
            percentage: '',
            total: 0
          },
          '106-120':{
            percentage: '',
            total: 0
          },
          '16-30': {
            percentage: '',
            total: 0
          },
          '31-45': {
            percentage: '',
            total: 0
          },
          '46-60': {
            percentage: '',
            total: 0
          },
          '61-75': {
            percentage: '',
            total: 0
          },
          '76-90': {
            percentage: '',
            total: 0
          },
          '91-105': {
            percentage: '',
            total: 0
          },
        },
      },
    },
    lineups: [],
    team: {
      id: 0,
      logo: '',
      name: ''
    }
  }
};

describe('<Result />', () => {
  it('should a render Result component', () => {
    renderTheme(<Result result={mockResult} />);

    const table = screen.getByRole('table');
    const heading = screen.getByRole('heading');
    const headerTable = screen.getByLabelText('Titulos da tabela');
    const bodyTable = screen.getByLabelText('Textos da tabela');

    expect(table).toBeInTheDocument();
    expect(headerTable).toBeInTheDocument();
    expect(bodyTable).toBeInTheDocument();
    expect(heading).toBeInTheDocument();
  });

  it('should not render Result component', () => {
    renderTheme(<Result result={newMock['response']} />);

    const alert = screen.getByLabelText('alert');

    expect(alert).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Result result={mockResult} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
