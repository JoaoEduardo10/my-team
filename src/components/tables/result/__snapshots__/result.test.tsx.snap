// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Result /> > should match snapshot 1`] = `
<div
  class="eEWGTn"
>
  <h3
    class="pCbkN"
  >
    Infomações De Partidas
  </h3>
  <table
    class="eceUpH"
    role="table"
  >
    <thead
      aria-label="Titulos da tabela"
      class="kpeZfO"
    >
      <tr
        class="hzbqHX"
      >
        <th
          class="jGFehp"
        >
          Formação mais utilizada
        </th>
        <th
          class="jGFehp"
        >
          Vitórias
        </th>
        <th
          class="jGFehp"
        >
          Derrotas
        </th>
        <th
          class="jGFehp"
        >
          Empates
        </th>
        <th
          class="jGFehp"
        >
          Total
        </th>
      </tr>
    </thead>
    <tbody
      aria-label="Textos da tabela"
      class="jIxYLu"
    >
      <tr
        class="hzbqHX"
      >
        <td
          class="kSUMkB"
        >
          4-5-2
        </td>
        <td
          class="kSUMkB"
        >
          6
        </td>
        <td
          class="kSUMkB"
        >
          5
        </td>
        <td
          class="kSUMkB"
        >
          8
        </td>
        <td
          class="kSUMkB"
        >
          29
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
